src/main/java/SearchEngine.java

```diff
@@ -0,0 +1,93 @@
+import strategies.impl.AllStrategyImpl;
+import strategies.impl.AnyStrategyImpl;
+import strategies.impl.NoneStrategyImpl;
+import strategies.SearchStrategy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+
+class SearchEngine {
+
+    private final FileReader fileReader;
```

Comment:
Do we really need to have file reader as a dependency here? It’s only used once

Reply:
fixed